lxd/response/response.go

```diff
@@ -568,7 +606,13 @@ func ManualResponse(hook func(w http.ResponseWriter) error) Response {
 }
 
 func (r *manualResponse) Render(w http.ResponseWriter, request *http.Request) error {
-	return r.hook(w)
+	err := r.hook(w)
+
+	if err == nil {
+		lxdRequest.MetricsCallback(request, metrics.Success)
+	}
```

Comment:
Should we be tracking this if the error is not nil? My understanding was that we need to mark all requests as completed at some point, otherwise the counter for ongoing requests will keep increasing.

Reply:
I believe any error returned by `Render` on the main LXD API should be handled like [this](https://github.com/hamistao/lxd/blob/improve_reboot_experience/lxd/daemon.go#L818), and thus the `Render` call on this line would mark the request as completed.

That said, it is missing a decrease on the ongoing request count in case that fails, I will fix this.